src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -4221,6 +4214,27 @@ static bool hasNonConstantInitializer(ImmutableArray<ImmutableArray<FieldOrPrope
             }
         }
 
+        private void AddSynthesizedTupleMembersIfNecessary(MembersAndInitializersBuilder builder, DeclaredMembersAndInitializers declaredMembersAndInitializers)
+        {
+            if (!this.IsTupleType)
+            {
+                return;
+            }
+
+            var synthesizedMembers = this.MakeSynthesizedTupleMembers(declaredMembersAndInitializers.NonTypeMembers);
```

Comment:
>declaredMembersAndInitializers.NonTypeMembers [](http://example.com/codeflow?start=70&length=45)

It feels like we should take synthesized members into account. Properties generated from primary constructors etc.

Reply:
If we wanted to do that, we'd have to reify/lock/cache all the previous synthesized members. I'd rather not add so many phases just to allow a record struct ValueTuple definition.

Reply:
> if we wanted to do that, we'd have to reify/lock/cache all the previous synthesized members

I don't think we can ovoid this. Prior to this change the process that was adding synthesized members was able to see tuple additions. We are changing the order, this means that we should see synthesized members here. Otherwise how can we avoid synthesizing duplicates, one there and the other one here?

Reply:
For example, try declaring a tuple type as a record struct with primary constructor.

Reply:
I guess we should be fine as long as we never synthesize user usable fields before we get here.